x-pack/solutions/observability/plugins/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -260,7 +260,12 @@ export const createCustomThresholdExecutor = ({
             [ALERT_GROUP]: groups,
             ...flattenAdditionalContext(additionalContext),
             ...getEcsGroups(groups),
+            ...groups.reduce((acc, curr) => {
+              const newValue = { ...acc, [curr.field]: curr.value };
+              return newValue;
+            }, {}),
```

Comment:
Add the grouping data to the alert's root.

Reply:
Perhaps we can specify the values within `runtimeFields` as a key/value pair and leave it out from here? This way we can keep the `payload` object strongly typed and merge the two behind the scenes in the framework.